File: python/tests/test_alarm_checker.py
"""
Tests for Alarm Checker module
"""

import pytest
from cps_utility_console.functions.alarm_checker import check_alarm


def test_alarm_ok():
    """Test value within range."""
    assert check_alarm(15, 10, 20) == "OK"


def test_alarm_low():
    """Test value below low threshold."""
    assert check_alarm(5, 10, 20) == "ALARM_LOW"


def test_alarm_high():
    """Test value above high threshold."""
    assert check_alarm(25, 10, 20) == "ALARM_HIGH"


def test_alarm_invalid_thresholds():
    """Test low >= high raises error."""
    with pytest.raises(ValueError, match="must be less than"):
        check_alarm(15, 20, 10)

File: cpp/tests/test_alarm_checker.cpp
#include <gtest/gtest.h>
#include "alarm_checker.hpp"

TEST(AlarmChecker, OK) {
    EXPECT_EQ(cps::checkAlarm(5.0f, 0.0f, 10.0f), "OK");
}

TEST(AlarmChecker, AlarmLow) {
    EXPECT_EQ(cps::checkAlarm(-1.0f, 0.0f, 10.0f), "ALARM_LOW");
}

TEST(AlarmChecker, AlarmHigh) {
    EXPECT_EQ(cps::checkAlarm(11.0f, 0.0f, 10.0f), "ALARM_HIGH");
}

TEST(AlarmChecker, InvalidThresholdThrows) {
    EXPECT_THROW(cps::checkAlarm(5.0f, 10.0f, 0.0f), std::invalid_argument);
}

TEST(AlarmChecker, EqualThresholdsThrow) {
    EXPECT_THROW(cps::checkAlarm(5.0f, 5.0f, 5.0f), std::invalid_argument);
}